src/components/Signal/Signal.tsx

```diff
@@ -85,21 +44,89 @@ const Signal: React.FC<SignalProps> = ({
       {...props}
```

Comment:
```suggestion
      style={{ overflow: 'visible', ...props.style}}
      {...props}
```
Overflow will prevent squishing icon if the container runs out of space